docs/reference/how-to/indexing-speed.asciidoc

```diff
@@ -0,0 +1,93 @@
+are known to perform better than spinning disks. Always use local storage,
+remote filesystems such as `NFS` or `SMB` should be avoided. Also beware of
+virtualized storage such as Amazon's `Elastic Block Storage`. Virtualized
+storage works very well with Elasticsearch, and it is appealing since it is so
+fast and simple to set up, but it is also unfortunately inherently slower on an
+ongoing basis when compared to dedicated local storage. In a recent informal
+test, even the highest performance provisioned IOPS SSD-backed EBS option was
+still substantially slower than the local instance-attached SSD, and remember
+that the local instance-attached SSD is still shared across all virtual machines
+on that physical machine so you will see otherwise inexplicable slowdowns if the
+other virtual machines on that physical machine suddenly become IO intensive.
+
+Stripe your index across multiple SSDs by configuring a RAID 0 array. Remember
+that it will increase the risk of failure since the failure of any one SSD
+destroys the index. However this is typically the right tradeoff to make:
+optimize single shards for maximum performance, and then add replicas across
+different nodes so there's redundancy for any node failures. You can also use
+<<modules-snapshots,snapshot and restore>> to backup the index for further
+insurance.
+
+=== Increase the refresh interval
+
+The default <<dynamic-index-settings,`index.refresh_interval`>> is `1s`, which
+forces elasticsearch to flush the indexing memory buffer every second.
+Increasing this value (to say, `30s`) will allow larger segments to flush and
+decreases future merge pressure.
+
+=== Disable refresh and replicas for initial loads
+
+If you need to load a large amount of data at once, you should disable refresh
+by setting `index.refresh_interval` to `-1` and set `index.number_of_replicas`
+to `0`. This will temporarily put your index at risk since the loss of any shard
+will cause data loss, but at the same time indexing will be faster since
+documents will be indexed only once. Once the initial loading is finished, you
+can set `index.refresh_interval` and `index.number_of_replicas` back to their
+original values.
+
+=== Indexing buffer size
+
+If your node is doing only heavy indexing, be sure
+<<indexing-buffer,`indices.memory.index_buffer_size`>> is large enough to give
+at most 512 MB indexing buffer per active shard (beyond that indexing performance
+does not typically improve). Elasticsearch takes that setting (a percentage of
+the java heap or an absolute byte-size), and divides it equally among the
+currently active shards on the node subject to `min_index_buffer_size` and
+`max_index_buffer_size` values; larger values means Lucene writes larger initial
+segments which reduces future merge pressure.
+
+The default is `10%` which is often plenty: for example, if you have 5 active
+shards on a node, and your heap is 25 GB, then each shard gets 1/5th of 10% of
```

Comment:
I think it is more fluid than that in master. Like they only take what they need.